src/computability/language.lean

```diff
@@ -47,16 +47,25 @@ def star (l : language α) : language α :=
 lemma star_def (l : language α) :
   l.star = { x | ∃ S : list (list α), x = S.join ∧ ∀ y ∈ S, y ∈ l} := rfl
 
-private lemma mul_assoc (l m n : language α) : (l * m) * n = l * (m * n) :=
+@[simp] lemma mem_one (x : list α) : x ∈ (1 : language α) ↔ x = [] := by refl
+@[simp] lemma mem_add (l m : language α) (x : list α) : x ∈ l + m ↔ x ∈ l ∨ x ∈ m :=
+by simp [add_def]
+lemma mem_mul (l m : language α) (x : list α) : x ∈ l * m ↔ ∃ a b, a ∈ l ∧ b ∈ m ∧ a ++ b = x :=
+by simp [mul_def]
+lemma mem_star (l : language α) (x : list α) :
+  x ∈ l.star ↔ ∃ S : list (list α), x = S.join ∧ ∀ y ∈ S, y ∈ l :=
+by refl
+
+private lemma mul_assoc_lang (l m n : language α) : (l * m) * n = l * (m * n) :=
 by { ext x, simp [mul_def], tauto {closer := `[subst_vars, simp *] } }
 
-private lemma one_mul (l : language α) : 1 * l = l :=
+private lemma one_mul_lang (l : language α) : 1 * l = l :=
 by { ext x, simp [mul_def, one_def], tauto {closer := `[subst_vars, simp [*]] } }
 
-private lemma mul_one (l : language α) : l * 1 = l :=
+private lemma mul_one_lang (l : language α) : l * 1 = l :=
 by { ext x, simp [mul_def, one_def], tauto {closer := `[subst_vars, simp *] } }
 
-private lemma left_distrib (l m n : language α) : l * (m + n) = (l * m) + (l * n) :=
+private lemma left_distrib_lang (l m n : language α) : l * (m + n) = (l * m) + (l * n) :=
```

Comment:
Any reason for renaming these? I'd assume these are just vestigial changes from your other now-merged PR

Reply:
They were renamed in the other PR as `mul_assoc` was causing naming conflicts in `star_mul_le_right_of_mul_le_right`